server/sonar-server/src/main/java/org/sonar/server/ce/ws/TaskFormatter.java

```diff
@@ -144,12 +142,25 @@ public TaskFormatter(DbClient dbClient, System2 system2) {
     return builder;
   }
 
-  private static Ce.Task.Builder setBranch(Ce.Task.Builder builder, String taskUuid, DtoCache componentDtoCache) {
-    componentDtoCache.getBranchName(taskUuid).ifPresent(
+  private static Ce.Task.Builder setBranchOrPullRequest(Ce.Task.Builder builder, String taskUuid, DtoCache componentDtoCache) {
+    componentDtoCache.getBranchKey(taskUuid).ifPresent(
       b -> {
-        builder.setBranch(b);
-        builder.setBranchType(componentDtoCache.getBranchType(taskUuid)
-          .orElseThrow(() -> new IllegalStateException(format("Could not find branch type of task '%s'", taskUuid))));
+        Common.BranchType branchType = componentDtoCache.getBranchType(taskUuid)
+          .orElseThrow(() -> new IllegalStateException(format("Could not find branch type of task '%s'", taskUuid)));
+        builder.setBranchType(branchType);
+        switch (branchType) {
+          case LONG:
+          case SHORT:
+            builder.setBranch(b);
+            break;
+          case PULL_REQUEST:
+            builder.setPullRequest(b);
+            // TODO set pull request title
+            builder.setPullRequestTitle(b);
```

Comment:
What about the url and the branch name of a pull request?

Reply:
not needed according to spec with @stas-vilchik 
do you confirm?

Reply:
I confirm that neither branch name, nor url is needed.